pkg/virt-controller/watch/vmi_test.go

```diff
@@ -3021,6 +3024,100 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			Expect(controller.cidsMap.reverse).To(BeEmpty())
 		})
 	})
+
+	Context("dynamic interface attachment", func() {
+		const vmName = "vm1"
+
+		var (
+			pod *k8sv1.Pod
+			vmi *virtv1.VirtualMachineInstance
+		)
+
+		expectPodStatusUpdateFailed := func(pod *k8sv1.Pod) {
+			kubeClient.Fake.PrependReactor("patch", "pods", func(action testing.Action) (handled bool, obj k8sruntime.Object, err error) {
+				_, ok := action.(testing.PatchAction)
+				Expect(ok).To(BeTrue())
+				return true, pod, fmt.Errorf("your cluster is doomed")
+			})
+		}
+
+		fakeHotPlugRequest := func(vmi *virtv1.VirtualMachineInstance, addOpts []virtv1.AddInterfaceOptions) {
+			for _, req := range addOpts {
+				vmi.Spec.Networks = append(
+					vmi.Spec.Networks,
+					virtv1.Network{
+						Name: req.InterfaceName,
+						NetworkSource: virtv1.NetworkSource{
+							Multus: &virtv1.MultusNetwork{NetworkName: req.NetworkName},
+						},
+					})
+			}
+		}
+
+		const (
+			firstVMInterface  = "oldIface1"
+			firstVMNetwork    = "oldnet1"
+			secondVMInterface = "oldiface2"
+			secondVMNetwork   = "oldnet2"
+		)
+
+		Context("k8s API is down - i.e. you cannot update the pod status", func() {
+			BeforeEach(func() {
+				vmi = newVMIWithOneNetwork(
+					newVMIWithOneNetwork(
+						api.NewMinimalVMI(vmName), firstVMNetwork, firstVMInterface),
+					secondVMNetwork, secondVMInterface)
+				pod = NewPodForVirtualMachine(vmi, k8sv1.PodRunning)
+				Expect(pod.Annotations).To(HaveKey(networkv1.NetworkAttachmentAnnot))
+				expectPodStatusUpdateFailed(pod)
+			})
+
+			DescribeTable("cannot handle dynamic network attachment", func(addOpts []virtv1.AddInterfaceOptions) {
+				fakeHotPlugRequest(vmi, addOpts)
+				Expect(controller.handleDynamicInterfaceRequests(vmi, pod)).To(HaveOccurred())
+			}, Entry("when adding an interface", []virtv1.AddInterfaceOptions{{
```

Comment:
Why there is a need for a table if there is only one scenario?

Reply:
Hm, I originally had more tests, but once I dropped everything related to unplug, this was left with a single test.

I'll turn it into an `It` container instead; good catch.